src/c/_cffi_backend.c

```diff
@@ -6109,6 +6108,18 @@ static void _my_PyErr_WriteUnraisable(PyObject *t, PyObject *v, PyObject *tb,
     if (extra_error_line == NULL)
         extra_error_line = "";
 
+#if PY_VERSION_HEX >= 0x030D0000
+    if (obj != NULL) {
+        PyErr_FormatUnraisable("Exception ignored %c%s%R%s",
+                               first_char, objdescr + 1, obj,
+                               extra_error_line);
+    }
+    else {
+        PyErr_FormatUnraisable("Exception ignored %c%s%s",
+                               first_char, objdescr + 1, extra_error_line);
+    }
+#else
+    PyObject *s;
```

Comment:
This does not compile on older Visual Studio versions, but I'm unsure how old these need to be.  You can't declare a variable anywhere else than at the start of a block in older versions of C.

Reply:
Ok, fixed. I made the assumption that cffi is built by a C99 compiler these days.

Reply:
I'm trying to keep the code compiling for 2.7, and that might require much older VS on Windows.

Reply:
I'm trying hard for provide a best effort support for Python 2.7 in my https://pythoncapi-compat.readthedocs.io/ project. I had C code to support VS2015 which doesn't support C99 `static inline`. But I removed it, it made the code harder to maintain. Moreover, GitHub Action doesn't provide Python 2.7. I'm now considering to officially remove Python 2 support and remove Python 2 code in pythoncapi-compat.

Reply:
Yep, I've tried "best effort" support like this so many times, but even with the best of intentions, it's effectively always broken when not backed by automated testing. It also ends up actively discouraging contributions (like this one!)...

Reply:
@nitzmahone Sure, I see the point, but in this case the final result is that the pull request was changed to 4 obvious lines instead of a more complex #if dance that missed originally a `PyErr_Restore()`, in addition to breaking a compiler (that may or may not be too old to consider).  I think it's a win for a project like CFFI.